packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -172,25 +192,101 @@ class BottomNavigationBar extends StatefulWidget {
   /// The index into [items] of the current active item.
   final int currentIndex;
 
+  /// The z-coordinate of this [BottomNavigationBar].
+  ///
+  /// If null, defaults to `8.0`.
+  ///
+  /// {@macro flutter.material.material.elevation}
+  final double elevation;
+
   /// Defines the layout and behavior of a [BottomNavigationBar].
   ///
   /// See documentation for [BottomNavigationBarType] for information on the meaning
   /// of different types.
   final BottomNavigationBarType type;
 
+  /// Deprecated, use [selectedItemColor].
```

Comment:
Why is this deprecated?

I'm confused about the relationship between fixedColor, selectedItemColor, and ThemeData.primaryColor. If fixedColor is set, that is selected over selectedItemColor. If fixedColor is unset, than ThemeData.primaryColor is used?

Reply:
So, `fixedColor` used to determine the color of the selected items, but _only_ in `Fixed` `BottomNavBar`s. In the case of `Shifting`, the selected items were _always_ white.

I'm updating this behavior so that you can customize the selected item color for *both* `Fixed` and `Shifting` bottom nav bars. 

My options were:
 1. Have `fixedColor` apply to both `Fixed` and `Shifting` bottom nav bars.
 2. Rename `fixedColor` to `selectedItemColor` <-- breaking change
 3. Create a new field called `selectedItemColor` and "deprecate" `fixedColor` since the name no longer makes sense.

So, I opted for the 3rd, but for backwards compatibility sake I made it so that if you don't supply `selectedItemColor` but you do supply a `fixedColor`, `fixedColor` will still be used.

Then it's a question of the defaults if *neither* parameter is provided. Again to keep this backwards compatible, I kept the behavior as is. That means that `Shifting` nav bars have a default selected item color of `Colors.White`, and `Fixed` bottom nav bars have a default selected item color of `ThemeData.primaryColor`.

Reply:
I don't think we should just comment on it as being deprecated unless we actually plan to remove it. 

cc @Hixie @matthew-carroll @goderbauer 

From https://github.com/flutter/flutter/issues/27753

It seems like this API is quite working as expected here. Do we take this as an opportunity to clean it up?

Reply:
Edit: is NOT quite working as expected

Reply:
I agree with Jonah here. There's a cost to having both colours now: the cognitive overhead of the API being more complicated, but also 8+ bytes per widget to store this field that we don't want people to use.

If you really wanted to keep it and not go through the deprecation process, you could have the constructor argument still exist and just have both it and selectedItemColor set the same field, and have the fixedColor field just return the same value.

But beyond that, we shouldn't say something is deprecated unless it's @Deprecated and going to be removed.

Reply:
Good point, if we do want to deprecate, I [should have used](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#clearly-mark-deprecated-apis) `@deprecated`. I'm fine either way, would people prefer to officially deprecate or leave both `fixedColor` and `selectedItemColor`?

Reply:
I think we should do any deprecations and removals outside of this PR to make it easier to land. I'm not sure what our policy is for the deprecated annotation, but an reasonable process might be:

- Land this PR with fixedColor pointing to selectedColor (and maybe an assert if you assign both).
- A breaking change request asking flutter dev about removing fixed color.
- At some later point in time, removing fixedColor in a separate PR

Reply:
Sounds good, I will follow the steps that Jonah laid out. Thanks for the input everyone.